src/mongocrypt-opts-private.h

```diff
@@ -55,6 +62,9 @@ typedef struct {
    void *log_ctx;
    _mongocrypt_buffer_t schema_map;
    _mongocrypt_opts_kms_provider_azure_t kms_provider_azure;
```

Comment:
Same note as with ctx, use a union for azure and gcp options

Reply:
These store KMS provider options. A user can configure multiple concurrent KMS providers and, say, do auto encryption with keys encrypted by different providers. Though the KMS provider options for creating a datakey should be represented with a union.